src/common/reactNativeProjectHelper.ts

```diff
@@ -10,53 +10,112 @@ import {Package} from "./node/package";
+                new Package(projectRoot).getPackageVersionFromNodeModules("react-native-windows")
+                    .then(version => ReactNativeProjectHelper.processVersion(version, false))
+                    .catch(err => "")
+            );
+        }
+
+        return Q.all(versionPromises).then(packageVersions => ({
+            reactNativeVersion: packageVersions[0] || "",
+            reactNativeWindowsVersion: packageVersions[1] || "",
+        }));
     }
 
-    public static getReactNativeVersionFromProjectPackage(cwd: string): Q.Promise<string> {
+    public static getReactNativeVersionsFromProjectPackage(cwd: string, isRNWindows: boolean = false): Q.Promise<RNPackageVersions> {
+        let parsedPackageNames: ParsedPackageName[] = [
+            {
+                packageName: "react-native",
+                useSemverCoerce: true,
+            },
+        ];
+
+        if (isRNWindows) {
+            parsedPackageNames.push({
+                packageName: "react-native-windows",
+                useSemverCoerce: false,
+            });
+        }
+
         const rootProjectPackageJson = new Package(cwd);
+
         return rootProjectPackageJson.dependencies()
             .then(dependencies => {
-                if (dependencies["react-native"]) {
-                    return ReactNativeProjectHelper.processVersion(dependencies["react-native"]);
-                }
                 return rootProjectPackageJson.devDependencies()
                     .then(devDependencies => {
-                        if (devDependencies["react-native"]) {
-                            return ReactNativeProjectHelper.processVersion(devDependencies["react-native"]);
-                        }
-                        return "";
+                        let versionPromises: Q.Promise<string>[] = [];
+                        parsedPackageNames.forEach(parsedPackageName => {
+                            versionPromises.push(
+                                Q.Promise<string>((resolve, reject) => {
+                                    if (dependencies[parsedPackageName.packageName]) {
+                                        resolve(ReactNativeProjectHelper.processVersion(dependencies[parsedPackageName.packageName], parsedPackageName.useSemverCoerce));
+                                    }
+                                    if (devDependencies[parsedPackageName.packageName]) {
+                                        resolve(ReactNativeProjectHelper.processVersion(devDependencies[parsedPackageName.packageName], parsedPackageName.useSemverCoerce));
+                                    }
+                                    resolve("");
+                                })
+                                .catch(err => "")
+                            );
+                        });
+
+                        return Q.all(versionPromises).then(packageVersions => ({
+                            reactNativeVersion: packageVersions[0],
```

Comment:
Please try to avoid constant indices because they can lead to the issues that are hard debuggable in such context.